### ISSUE TITLE
reducer: replace VariablesDeclaration with expression

### ISSUE THREAD

**Issue by paulthomson** [google/graphicsfuzz#480]
When not preserving semantics, the reducer should try:

```glsl
void main()
{
 bool doSwap = checkSwap();
}
```

```glsl
void main()
{
 checkSwap();
}
```

**Comment by paulthomson** [comment]
Could do this?

```glsl
bool doSwap = checkSwap();
->
bool doSwap; doSwap = checkSwap();
```

**Comment by paulthomson** [comment]
int a = 42, b, c = foo(), d;
->
int a, b, c, d;
a = 42;
c = foo();

**Comment by paulthomson** [comment]
This can't be semantics-preserving, becuase e.g. we can do:

int a = 12, b = a;

Or:

int a = foo(), b = bar(); // foo and bar have side effects; the effects of bar depend on what foo did

**Comment by afd** [comment]
Fixed via #529 and #576.
